File: BE/src/models/OrderedProduct.js
const mongoose = require('mongoose');

const OrderedProductSchema = new mongoose.Schema({
    ProductCode: { type: String, required: true, unique: true },
    Name: { type: String, required: true },
    Quantity: { type: Number, required: true },
    Price: { type: Number, required: true },
});

const OrderedProduct = mongoose.model('orderedProduct', OrderedProductSchema, 'orderedProduct');
module.exports = OrderedProduct;

File: App/Model/OrderedProduct.cs
using System;
using System.Collections.Generic;
using System.ComponentModel;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace App.Model
{
    public class OrderedProduct : INotifyPropertyChanged
    {
        public event PropertyChangedEventHandler? PropertyChanged;

        public string ProductCode { get; set; }
        public string Name { get; set; }
        public int Quantity { get; set; }
        public decimal Price { get; set; }

        public OrderedProduct(string productCode, string name, int quantity, decimal price)
        {
            ProductCode = productCode;
            Name = name;
            Quantity = quantity;
            Price = price;
        }

        public OrderedProduct() { }
    }
}